x.c

```diff
@@ -1663,10 +1885,48 @@ xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len, int x, i
 		XftDrawGlyphFontSpec(xw.draw, fg, specs, len);
 	}
 
-	/* Render underline and strikethrough. */
+	/* Render underline/undercurl and strikethrough. */
```

Comment:
i think we could still factor all this out into one more "drawunderline" function, or just combine those two functions curls/underline into one

Reply:
While that's valid I think it would be a bad idea to have such function that is essentially a
```c
void drawunderlineorundercurl() { ... }
```
since it will needs so much unrelated stuff passed into it but needed because a single property of an entire struct is needed.
a less severe refactor that follows the code logic could be the correct move, take a look at the latest commit as the whole undercurl logic is 11 lines now.